core/trino-main/src/main/java/io/trino/execution/DistributionSnapshot.java

```diff
@@ -0,0 +1,255 @@
+    private final double p10;
+    private final double p25;
+    private final double p50;
+    private final double p75;
+    private final double p90;
+    private final double p95;
+    private final double p99;
+
+    public static List<OperatorStats> trimOperatorStats(List<OperatorStats> operatorStats)
+    {
+        requireNonNull(operatorStats, "operatorStats is null");
+        return operatorStats.stream()
+                .map(DistributionSnapshot::trimOperatorStats)
+                .collect(toImmutableList());
+    }
+
+    public static OperatorStats trimOperatorStats(OperatorStats operatorStats)
+    {
+        requireNonNull(operatorStats, "operatorStats is null");
+        return new OperatorStats(
+                operatorStats.getStageId(),
+                operatorStats.getPipelineId(),
+                operatorStats.getOperatorId(),
+                operatorStats.getPlanNodeId(),
+                operatorStats.getOperatorType(),
+                operatorStats.getTotalDrivers(),
+                operatorStats.getAddInputCalls(),
+                operatorStats.getAddInputWall(),
+                operatorStats.getAddInputCpu(),
+                operatorStats.getPhysicalInputDataSize(),
+                operatorStats.getPhysicalInputPositions(),
+                operatorStats.getPhysicalInputReadTime(),
+                operatorStats.getInternalNetworkInputDataSize(),
+                operatorStats.getInternalNetworkInputPositions(),
+                operatorStats.getRawInputDataSize(),
+                operatorStats.getInputDataSize(),
+                operatorStats.getInputPositions(),
+                operatorStats.getSumSquaredInputPositions(),
+                operatorStats.getGetOutputCalls(),
+                operatorStats.getGetOutputWall(),
+                operatorStats.getGetOutputCpu(),
+                operatorStats.getOutputDataSize(),
+                operatorStats.getOutputPositions(),
+                operatorStats.getDynamicFilterSplitsProcessed(),
+                trimMetrics(operatorStats.getMetrics()),
+                trimMetrics(operatorStats.getConnectorMetrics()),
+                trimMetrics(operatorStats.getPipelineMetrics()),
+                operatorStats.getPhysicalWrittenDataSize(),
+                operatorStats.getBlockedWall(),
+                operatorStats.getFinishCalls(),
+                operatorStats.getFinishWall(),
+                operatorStats.getFinishCpu(),
+                operatorStats.getUserMemoryReservation(),
+                operatorStats.getRevocableMemoryReservation(),
+                operatorStats.getPeakUserMemoryReservation(),
+                operatorStats.getPeakRevocableMemoryReservation(),
+                operatorStats.getPeakTotalMemoryReservation(),
+                operatorStats.getSpilledDataSize(),
+                operatorStats.getBlockedReason(),
+                operatorStats.getInfo());
+    }
+
+    private static Metrics trimMetrics(Metrics metrics)
```

Comment:
I don't like a word trim in that context. Maybe "pruned" would be better as we use it in similar context in other places?